File: src/fe-hunt-editor/src/lib/models/Hint.ts
/**
 * Represents a hint for an assignment.
 */
export interface Hint {
	/** The type of the hint. */
	hintType: HintType;
	/** The data associated with the hint. */
	data: string;
	/** Optional additional data for the hint. */
	additionalData?: string;
}

/**
 * Enum for the different types of hints.
 */
export enum HintType {
	Text,
	Image,
	Video,
	Audio
}

File: old-files/src/be-hunt-api/Hunts.Domain/Entities/Hint.cs
﻿namespace Hunts.Domain.Entities
{
    /// <summary>
    /// Represents a hint for an assignment.
    /// </summary>
    public sealed class Hint
    {
        public int Id { get; set; }
        public HintType HintType { get; set; }
        public string Data { get; set; } = default!;    
        public string? additionalData {get; set;}
        public int AssignmentId { get; set; }
        public required Assignment Assignment { get; set; }
    }
}
